### PR TITLE
Add links to Docker images, update documentation

### PR DESCRIPTION
**Pull request by yurishkuro**

### CODE REVIEW

File: docs/deployment.md [hunk 1]
```diff
@@ -1,48 +1,67 @@
...
+14267 | TChannel | used by **jaeger-agent** to send spans in jaeger.thrift format
+14268 | HTTP     | can accept spans directly from clients in Jaeger or Zipkin Thrift 
+
 
-Many instances of collectors can be run in parallel.
+## Storage Backend
 
-## Cassandra
+Collectors require a persistent storage backend. Cassandra 3.x is the primary supported storage.
+There is ongoing work to add support for ElasticSearch, MySQL, and ScyllaDB.
```
Comment (jpkrohling) [inline docs/deployment.md]: `s/ElasticSearch/Elasticsearch/gi`. Yeah, I know...

File: docs/deployment.md [hunk 2]
```diff
@@ -1,48 +1,67 @@
 # Deployment
 
-*This section is under construction.*
+The main Jaeger backend components are released as Docker images on Docket Hub:
```
Comment (vprithvi) [inline docs/deployment.md]: s/Docket/Docker

File: docs/deployment.md [hunk 3]
```diff
@@ -1,48 +1,67 @@
...
 
-We will soon provide Docker images for individual Jaeger components
-[Issue #157](https://github.com/uber/jaeger/pull/157),
-as well as orchestration templates, for example to run Jaeger on
-[Kubernetes](https://github.com/jaegertracing/jaeger-kubernetes)
-and [OpenShift](https://github.com/jaegertracing/jaeger-openshift).
+Component             | Repository
+--------------------- | ---
+**jaeger-agent**      | [https://hub.docker.com/r/jaegertracing/jaeger-agent/](https://hub.docker.com/r/jaegertracing/jaeger-agent/)
```
Comment (vprithvi) [inline docs/deployment.md]: We could remove `https://` from the text we display for the link to reduce clutter

File: docs/deployment.md [hunk 4]
```diff
@@ -1,48 +1,67 @@
...
 
-Collectors require almost no configuration except for the location of Cassandra cluster.
+Many instances of **jaeger-collector** can be run in parallel.
+Collectors require almost no configuration, except for the location of Cassandra cluster,
+via `-cassandra.keyspace` and `-cassandra.servers` options.
 All configuration parameters can be provided via command line options.
-At default settings the collector exposes the following ports: `14267 14268`.
 
 ```go
 go run ./cmd/collector/main.go -h
```
Comment (objectiser) [inline docs/deployment.md]: Would it be better to use:
`docker run -it --rm --network=host jaegertracing/jaeger-collector /go/bin/collector-linux -h`
so does not require user to run a Golang app?

Comment (yurishkuro) [inline docs/deployment.md]: @objectiser I added docker cmd as alternative. I think `--network=host` is not needed since it just prints help and exist.

**Pull request merged**